### PR TITLE
fix: Members Should not Be Able to See Org Admin Settings

### LINKED ISSUE #14606: [CAL-3474] Members Should not Be Able to See Org Admin Settings
Members can see admin settings on the left hand side

<sub>From [SyncLinear.com](https://synclinear.com) | [CAL-3474](https://linear.app/calcom/issue/CAL-3474/members-should-not-be-able-to-see-org-admin-settings)</sub>

### PR DESCRIPTION
**Pull request by Amit91848**
## What does this PR do?

Fixes #14606 

For members

Screenshot 2024-04-16 at 10 07 49 PM

For Admin Or Owner

Screenshot 2024-04-16 at 10 08 15 PM

## Type of change

- Bug fix (non-breaking change which fixes an issue)

## Mandatory Tasks

- [x] Make sure you have self-reviewed the code. A decent size PR without self-review might be rejected.

### CODE REVIEW

File: packages/features/settings/layouts/SettingsLayout.tsx [hunk 1]
```diff
@@ -257,6 +257,7 @@ const SettingsSidebarContainer = ({
 
   const isOrgAdminOrOwner =
     currentOrg && currentOrg?.user?.role && ["OWNER", "ADMIN"].includes(currentOrg?.user?.role);
+  let bottomMargin = false;
```
Comment (zomars) [inline packages/features/settings/layouts/SettingsLayout.tsx:260]: Why is this needed?

File: packages/features/settings/layouts/SettingsLayout.tsx [hunk 2]
```diff
@@ -312,19 +313,30 @@ const SettingsSidebarContainer = ({
...
-                          tab.children && index === tab.children?.length - 1 && "!mb-3"
-                        }`}
-                        disableChevron
-                      />
-                    ))}
+                    {tab.children?.map((child, index) => {
+                      if (tab.href === "/settings/organizations" && !isOrgAdminOrOwner) {
+                        // Only show profile and members for user role.
+                        if (!["profile", "members"].includes(child.name)) {
+                          return;
```
Comment (zomars) [inline packages/features/settings/layouts/SettingsLayout.tsx:320]:
```suggestion
                          return null;
```

Comment (zomars) [reply packages/features/settings/layouts/SettingsLayout.tsx:320]: We need to return a valid React Node

File: packages/features/settings/layouts/SettingsLayout.tsx (hunk 1 above)
Comment (Amit91848) [reply packages/features/settings/layouts/SettingsLayout.tsx:260]: Margin was only being applied on the last child of the Tab

https://github.com/calcom/cal.com/blob/ae64d7ba8223c2d2992a4db9f003a1c94efa869f/packages/features/settings/layouts/SettingsLayout.tsx#L315-L327

We set it to true of in case of Org Members to also add margin there

https://github.com/calcom/cal.com/blob/fb92d0c827bddf06c639e6c9831a631da3d187be/packages/features/settings/layouts/SettingsLayout.tsx#L322-L324

https://github.com/calcom/cal.com/blob/fb92d0c827bddf06c639e6c9831a631da3d187be/packages/features/settings/layouts/SettingsLayout.tsx#L333-L335

Screenshot 2024-04-16 at 10 07 49 PM

File: packages/features/settings/layouts/SettingsLayout.tsx (hunk 2 above)
Comment (Amit91848) [reply packages/features/settings/layouts/SettingsLayout.tsx]: Fixed this.

File: packages/features/settings/layouts/SettingsLayout.tsx [hunk 3]
```diff
@@ -312,19 +313,30 @@ const SettingsSidebarContainer = ({
...
+                    {tab.children?.map((child, index) => {
+                      if (tab.href === "/settings/organizations" && !isOrgAdminOrOwner) {
+                        // Only show profile and members for user role.
+                        if (!["profile", "members"].includes(child.name)) {
+                          return null;
+                        }
+                        if (child.name === "members") {
+                          bottomMargin = true;
+                        }
+                      }
```
Comment (Udit-takkar) [inline packages/features/settings/layouts/SettingsLayout.tsx:325]: instead of this can you make changes inside useTabs() hook and remove the tabs that are not required for 'USER' role.

Comment (Udit-takkar) [reply packages/features/settings/layouts/SettingsLayout.tsx:325]: you won't need bottomMargin this way

Comment (Amit91848) [reply packages/features/settings/layouts/SettingsLayout.tsx:325]: Yeah that's better, will make changes.

**Review by Udit-takkar — DISMISSED** [review]
LGTM. pushed some changes

**Comment by Amit91848** [comment]
Had merge conflicts.

**Review by Udit-takkar — COMMENTED** [review]
@Amit91848 thanks for your contribution but we'll be going forward with sean's PR https://github.com/calcom/cal.com/pull/14666
